### DISCUSSION TITLE
Package doesn't include generated files into project (Not usable)

### DISCUSSION THREAD

**Discussion by smaillet** [microsoft/CsWin32#411] [Q&A]
## Actual behavior
The generated files are not included in the project and are therefore unusable to the application code.

## Expected behavior

Like any other generator I expect the generated output to automatically be included as code in the project.
Additionally, I'd expect to be able to set the full namespace of the generated content. (It seems quite limited at the moment and always ends up including a nested 'PInvoke' namesapce I don't want.

## Repro steps

1. `NativeMethods.txt` content:
```
SetWindowPos
```

2. `NativeMethods.json` content (if present):
```json
```
[NOTE NativeMethods.json is, at present an undocumented thing... WHat is it? Better docs would help a lOT, one GIF doesn't tell a clear story]

3. Any of your own code that should be shared?

### Context

- CsWin32 version: [e.g. `0.1.560-beta`]
- Target Framework: [e.g. `net5.0-windows10.0.19041.0`]

**Comment by AArnott** [comment]
> NativeMethods.json is, at present an undocumented thing

It is documented. [The readme introduces it](https://github.com/microsoft/CsWin32#customizing-generated-code) and if you use it as it guides you, then a decent JSON editor (including VS) will use the `$schema` property to give you auto-completion and intellisense documentation on the options available.

> The generated files are not included in the project and are therefore unusable to the application code.

CsWin32 is implemented as a roslyn Source Generator and has no say in how the generated files are included in the project. This is a feature of the C# compiler and its integration with VS and msbuild. If this isn't working for you, my guess is there's something not configured right in your setup somehow or you're hitting a bug. 

If you can include a build log (with `dotnet build -bl`) of your project's failed build we might be able to help here. But getting support from dotnet/roslyn *may* be where we send you if it's not a CsWin32-specific thing.
